jvb/src/main/kotlin/org/jitsi/videobridge/cc/av1/Av1DDQualityFilter.kt

```diff
@@ -0,0 +1,473 @@
+             * current DT.
+             */
+            val frameInfo = frame.frameInfo ?: run {
+                needsKeyframe = true
+                return@doAcceptFrame false
+            }
+            var currentDt = getDtFromIndex(currentIndex)
+            val externalTargetDt = if (currentEncoding == externalTargetEncoding) {
+                getDtFromIndex(externalTargetIndex)
+            } else {
+                currentDt
+            }
+
+            if (
+                frame.activeDecodeTargets != null &&
+                !frame.activeDecodeTargets.containsDecodeTarget(externalTargetDt)
+            ) {
+                /* This shouldn't happen, because we should have set layeringChanged for this packet. */
+                logger.warn {
+                    "External target DT $externalTargetDt not present in current decode targets 0x" +
+                        Integer.toHexString(frame.activeDecodeTargets) + " for frame $frame."
+                }
+                return false
+            }
+
+            if (currentDt != externalTargetDt) {
+                val frameMap = av1FrameMap[frame.ssrc]
+                if (frameInfo.dti.getOrNull(externalTargetDt) == null) {
+                    logger.warn { "Target DT $externalTargetDt not present for frame $frame [frameInfo $frameInfo]" }
+                }
+                if (frameInfo.dti[externalTargetDt] == DTI.SWITCH &&
+                    frameMap != null &&
+                    frameInfo.fdiff.all {
+                        frameMap.getIndex(frame.index - it)?.isAccepted == true
+                    }
+                ) {
+                    logger.debug { "Switching to DT $externalTargetDt from $currentDt" }
+                    currentDt = externalTargetDt
+                    currentIndex = externalTargetIndex
+                } else {
+                    if (frame.structure?.canSwitchWithoutKeyframe(currentDt, externalTargetDt) != true) {
+                        logger.debug {
+                            "Want to switch to DT $externalTargetDt from $currentDt, requesting keyframe"
+                        }
+                        needsKeyframe = true
+                    }
+                }
+            }
+
+            val currentFrameDti = frameInfo.dti[currentDt]
+            if (currentEncoding > externalTargetEncoding) {
+                (currentFrameDti == DTI.SWITCH || currentFrameDti == DTI.REQUIRED)
+            } else {
+                (currentFrameDti != DTI.NOT_PRESENT)
+            }
+        } else {
+            // In this branch we're not processing a keyframe and the
+            // currentEncoding is in suspended state, which means we need
+            // a keyframe to start streaming again. Reaching this point also
+            // means that we want to forward something (because both
+            // externalEncodingTarget is not suspended) so we set the request keyframe flag.
```

Comment:
That comment has been there since George rewrote the VP8 filter in 2019, and then copied over to VP9 and AV1, without anyone noticing it was wrong before.